pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -86,11 +87,25 @@ def zyz_decomposition(U, wire):
         omega = 2 * math.angle(U[1, 1])
         return [qml.RZ(omega, wires=wire)]
 
-    # If not diagonal, compute the angle of the RY
+    # Check if the matrix is off-diagonal. This means we can express it in terms
+    # of a single Z rotation and a Y rotation of -pi.
+    if math.allclose(U[0, 0], [0.0]):
```

Comment:
Does this also imply that `U[1, 1] = 0`?

Reply:
Yup, since you can always write a unitary like this:

![image](https://user-images.githubusercontent.com/2068515/133078501-125f2acf-225d-4c2c-9e73-b8cd0504b79a.png)

where a, b are complex numbers, so if a = 0 it must be that its real part is zero (since there's no way to set e^ix = 0), therefore U[1, 1] is 0 as well.

Reply:
thanks @glassnotes!